src/control/server/storage/scm/ipmctl.go

```diff
@@ -346,33 +352,38 @@ func parseShowRegionOutput(log logging.Logger, text string) (_ storage.ScmState,
 			if !appDirect {
 				continue
 			}
-			c, err := humanize.ParseBytes(kv[1])
+			fc, err := humanize.ParseBytes(kv[1])
 			if err != nil {
-				return storage.ScmStateUnknown, err
+				return nil, storage.ScmStateUnknown, err
 			}
-			totalFree += c
+			regionFreeBytes = append(regionFreeBytes, fc)
 		}
 		appDirect = false
 	}
 
-	if totalFree > 0 {
-		log.Debugf("PMem regions have %s free capacity", humanize.Bytes(totalFree))
-		return storage.ScmStateFreeCapacity, nil
+	if len(regionFreeBytes) == 0 {
+		return nil, storage.ScmStateUnknown, errors.New("no AppDirect PMem regions parsed")
+	}
+	for _, freeBytes := range regionFreeBytes {
+		if freeBytes > 0 {
+			log.Debugf("PMem regions have %v free bytes", regionFreeBytes)
+			return regionFreeBytes, storage.ScmStateFreeCapacity, nil
+		}
 	}
 	log.Debug("PMem regions have no free capacity")
-	return storage.ScmStateNoFreeCapacity, nil
+	return regionFreeBytes, storage.ScmStateNoFreeCapacity, nil
 }
 
-func (cr *cmdRunner) getRegionState() (storage.ScmState, error) {
+func (cr *cmdRunner) getRegionState() ([]uint64, storage.ScmState, error) {
```

Comment:
This addition feels like it violates the single responsibility principle. Plus it seems like the new value is only used in one place. `getRegionState` could be left as it was, and restructure the area where we need to get the region free capacity out of the "show -region" command.

Reply:
done, I think it's improved now, thank you